doc/source/user_guide/tutorials/mesh/create_a_mesh_from_scratch.rst

```diff
@@ -0,0 +1,158 @@
+--------------------------------
+
+To create a mesh you must define the nodes connectivity. This means to define
+the nodes ids connected to each element.
+
+Here, we create a function that will find this connectivity.
+
+.. jupyter-execute::
+
+    def search_sequence_numpy(arr, seq):
+        """Find a sequence in an array and return its index."""
+        indexes = np.where(np.isclose(arr, seq[0]))
+        for index in np.nditer(indexes[0]):
+            if index % 3 == 0:
+                if np.allclose(arr[index + 1], seq[1]) and np.allclose(arr[index + 2], seq[2]):
+                    return index
+        return -1
+
+Add nodes
+---------
+
+Add |Nodes| to the |MeshedRegion| object.
+
+.. jupyter-execute::
+
+    node_id = 1
+    for i, x in enumerate(
+        [float(i) * length / float(num_nodes_in_length) for i in range(0, num_nodes_in_length)]
+    ):
+        for j, y in enumerate(
+            [float(i) * width / float(num_nodes_in_width) for i in range(0, num_nodes_in_width)]
+        ):
+            for k, z in enumerate(
+                [float(i) * depth / float(num_nodes_in_depth) for i in range(0, num_nodes_in_depth)]
+            ):
+                my_meshed_region.nodes.add_node(node_id, [x, y, z])
+                node_id += 1
+
+Get the nodes coordinates field.
+
+.. jupyter-execute::
+
+    my_nodes_coordinates = my_meshed_region.nodes.coordinates_field
+
+Set the mesh properties
+-----------------------
+
+Set the mesh unit.
+
+.. jupyter-execute::
+
+    my_meshed_region.unit = "mm"
+
+Set the nodes coordinates.
+
+.. jupyter-execute::
+
+    # Get the nodes coordinates data
+    my_nodes_coordinates_data = my_nodes_coordinates.data
+    # As we use the connectivity function we need to get the data as a list
+    my_nodes_coordinates_data_list = my_nodes_coordinates.data_as_list
+    # Set the nodes scoping
+    my_coordinates_scoping = my_nodes_coordinates.scoping
+
+Add elements
+------------
+Add |Elements| to the |MeshedRegion| object.
+
+.. jupyter-execute::
+
+    element_id = 1
+    for i, x in enumerate(
+        [float(i) * length / float(num_nodes_in_length) for i in range(num_nodes_in_length - 1)]
+    ):
+        for j, y in enumerate(
+            [float(i) * width / float(num_nodes_in_width) for i in range(num_nodes_in_width - 1)]
+        ):
+            for k, z in enumerate(
+                [float(i) * depth / float(num_nodes_in_depth) for i in range(num_nodes_in_depth - 1)]
+            ):
+                coord1 = np.array([x, y, z])
+                connectivity = []
+                for xx in [x, x + length / float(num_nodes_in_length)]:
+                    for yy in [y, y + width / float(num_nodes_in_width)]:
+                        for zz in [z, z + depth / float(num_nodes_in_depth)]:
+                            data_index = search_sequence_numpy(my_nodes_coordinates_data_list, [xx, yy, zz])
+                            scoping_index = int(data_index / 3)  # 3components
```

Comment:
If you choose to go with the previous suggestion, then you just need to do this:
```suggestion
                            scoping_index = search_sequence_numpy(my_nodes_coordinates_data, [xx, yy, zz])
```

Reply:
Of course, you would still need to take care of the suggestions already given by Jorge.